radix-engine/src/blueprints/epoch_manager/epoch_manager.rs

```diff
@@ -287,6 +351,47 @@ impl EpochManagerBlueprint {
         Ok((validator_address, owner_token_bucket))
     }
 
+    fn update_proposal_statistics<Y>(
+        progressed_rounds: u64,
+        proposal_history: LeaderProposalHistory,
+        api: &mut Y,
+    ) -> Result<(), RuntimeError>
+    where
+        Y: ClientApi<RuntimeError>,
+    {
+        if proposal_history.gap_round_leaders.len() as u64 != progressed_rounds - 1 {
```

Comment:
Minor: feels like a little bit of a code smell that `progressed_rounds` is only used for verification. Maybe makes more sense for the check to occur before this function is called?

Reply:
I agree it smells a bit.

But at the other hand, doing a `proposal_history` argument validation fits very well inside the `update_proposal_statistics()` function (which is supposed to process the `proposal_history`). I am reluctant to delegate the "argument validation" responsibility to the caller of the function (it would be more visible if we were e.g. calling this function from 2 places).